src/Functions/FunctionBinaryArithmetic.h

```diff
@@ -1176,6 +1190,93 @@ class FunctionBinaryArithmetic : public IFunction
         return column_to;
     }
 
+    ColumnPtr executeDateTime64Subtraction(const ColumnsWithTypeAndName & arguments, const DataTypePtr & result_type, size_t input_row_count) const
+    {
+        using OpImpl = DecimalBinaryOperation<Op, Decimal64, false>;
```

Comment:
```suggestion
        using OpImpl = DecimalBinaryOperation<Op, Decimal64, /* check_overflow */ false>;
```

Why `false` btw?

Reply:
I checked the range of DateTime64, yes we need to check overflow. I've updated the code and test case to cover it, thanks